addon/helpers/compact.js

```diff
@@ -0,0 +1,66 @@
+import Ember from 'ember';
+
+const {
+  Helper,
+  isArray,
+  defineProperty,
+  get,
+  set,
+  observer,
+  isPresent,
+  computed,
+  copy,
+  typeOf,
+  A: emberArray
+} = Ember;
+const keys = Object.keys || Ember.keys;
+
+// For compacting object instances
+export function compact(objectInstance) {
+  return keys(objectInstance).reduce((compactedObject, key) => {
+    let val = objectInstance[key];
+    if (isPresent(val)) {
+      compactedObject[key] = val;
+    }
+    return compactedObject;
+  }, {});
+}
+
+export default Helper.extend({
+  compute([arrayOrObject]) {
+    set(this, 'arrayOrObject', arrayOrObject);
+
+    if (isArray(arrayOrObject)) {
+      set(this, 'keys', []);
+    } else if (typeOf(arrayOrObject) === 'object') {
+      set(this, 'keys', keys(arrayOrObject));
+    } else {
+      return emberArray([arrayOrObject]);
+    }
+
+    return get(this, 'content');
+  },
+
+  arrayOrObject: null,
+  keys: null,
+
+  keysToWatchDidChange: observer('keys', function() {
```

Comment:
I don't think we should watch for the object his keys at all. Keys added to the object will not be watched, which will result in all kinds of weird behaviour. Let's keep it simple.